package/MDAnalysis/coordinates/RDKit.py

```diff
@@ -95,10 +134,591 @@ def __init__(self, filename, **kwargs):
+        expected_vs = PERIODIC_TABLE.GetValenceList(atom.GetAtomicNum())
+        current_v = atom.GetTotalValence() - atom.GetFormalCharge()
+        nue = [v - current_v for v in expected_vs]
+        # if there's only one possible valence state and the corresponding
+        # NUE is negative, it means we can only add a positive charge to
+        # the atom
+        if (len(nue) == 1) and (nue[0] < 0):
+            atom.SetFormalCharge(-nue[0])
+            mol.UpdatePropertyCache(strict=False)
+        # go to next atom if above case or atom has no unpaired electron
+        if (len(nue) == 1) and (nue[0] <= 0):
+            continue
+        else:
+            neighbors = atom.GetNeighbors()
+            # check if one of the neighbors has a common NUE
+            for i, na in enumerate(neighbors, start=1):
+                # get NUE for the neighbor
+                na_expected_vs = PERIODIC_TABLE.GetValenceList(
+                    na.GetAtomicNum())
+                na_current_v = na.GetTotalValence() - na.GetFormalCharge()
+                na_nue = [v - na_current_v for v in na_expected_vs]
+                # smallest common NUE
+                common_nue = min(
+                    min([i for i in nue if i >= 0], default=0),
+                    min([i for i in na_nue if i >= 0], default=0)
+                )
+                # a common NUE of 0 means we don't need to do anything
+                if common_nue != 0:
+                    # increase bond order
+                    bond = mol.GetBondBetweenAtoms(
+                        atom.GetIdx(), na.GetIdx())
+                    order = common_nue + 1
+                    bond.SetBondType(RDBONDORDER[order])
+                    mol.UpdatePropertyCache(strict=False)
+                    if i < len(neighbors):
+                        # recalculate nue for atom
+                        current_v = atom.GetTotalValence()
+                        nue = [v - current_v for v in expected_vs]
+
+            # if the atom still has unpaired electrons
+            current_v = atom.GetTotalValence() - atom.GetFormalCharge()
+            nue = [v - current_v for v in expected_vs][0]
+            if nue > 0:
+                # keep the radical if it's a terminal atom
+                # else transform it to a negative charge
+                if atom.GetIdx() not in terminal_atom_indices:
+                    atom.SetFormalCharge(-nue)
+                    atom.SetNumRadicalElectrons(0)
+                    mol.UpdatePropertyCache(strict=False)
+
+
+def _standardize_patterns(mol):
+    """Standardizes functional groups
+
+    Uses :func:`_rebuild_conjugated_bonds` to standardize conjugated systems,
+    and SMARTS reactions for other functional groups.
+    Due to the way reactions work, we first have to split the molecule by
+    fragments. Then, for each fragment, we apply the standardization reactions.
+    Finally, the fragments are recombined.
```

Comment:
I think I've finally wrapped my head around this. Here we will need to document a list of all the different reactions we are enforcing to happen. Users will need to be aware of it, in case it happens that they have a system with a feature that is unintentionally converted by one of these reactions.